File: src/main/java/com/hyungu/whoseturn/entities/RoomEntity.java
package com.hyungu.whoseturn.entities;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "ROOMS")
public class RoomEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "ROOM_ID", nullable = false)
    private String id;

    @ManyToMany(mappedBy = "ROOM_ID")
    private List<RoomUserEntity> roomUsers;

    public RoomEntity() {
    }

    // getter
    public String getId() {
        return id;
    }

}

File: src/main/kotlin/com/hyungu/whoseturn/entities/RoomEntity.kt
package com.hyungu.whoseturn.entities

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany

@Entity(name = "rooms")
data class RoomEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: String?,

    @ManyToMany
    @JoinTable(name = "room_user")
    @JoinColumn(name = "id", referencedColumnName = "room_id")
    var users: List<UserEntity>
)

